misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -1186,12 +1186,16 @@ public XMLNamespaceDeclarationNode transform(XMLNamespaceDeclarationNode xMLName
         Token asKeyword = getToken(xMLNamespaceDeclarationNode.asKeyword().orElse(null));
         IdentifierToken namespacePrefix = this.modifyNode(xMLNamespaceDeclarationNode.namespacePrefix().orElse(null));
         Token semicolonToken = getToken(xMLNamespaceDeclarationNode.semicolonToken());
+        if (asKeyword != null || namespacePrefix != null) {
```

Comment:
Shouldn't this be an AND expression here? Since both the `namespacePrefix` and the `asKeyword` should not be null.

Reply:
The purpose of using the `OR` is, if either one of `asKeyword` or `namespacePrefix` is available (not null), then the _if_ condition would be true and both of them should be present when applying the modifications. 
Otherwise if even a single one out of them is null, there would be an assertion error generated when checking in the `Objects.requireNonNull()` methods of _tree node modifier utility_. 

And, even if the `asKeyword` is `null`, still the parser would be returning a `MISSING[as]` token if the `namespacePrefix` is present. 

Is this approach correct, or, are there any other suggestions?

Reply:
@IrushiL, Thank you for pointing out the issue here. Fixed the logical error.